lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -8310,6 +8327,16 @@ namespace Js
             return false;
         }
 
+     	bool evictCachedType = true;
+        if (DynamicType::Is(type->GetTypeId()))
+        {
+            DynamicType* dynamicType = static_cast<DynamicType*>(type);
+            if ((dynamicType->GetTypeHandler()->IsSimpleDictionaryTypeHandler() || dynamicType->GetTypeHandler()->IsDictionaryTypeHandler()) &&
```

Comment:
>(dynamicType->GetTypeHandler()->IsSimpleDictionaryTypeHandler() || dynamicType->GetTypeHandler()->IsDictionaryTypeHandler()) [](start = 16, length = 124)

Why not just have condition for non-shared type. BTW, I have similar changes in [https://github.com/Microsoft/ChakraCore/pull/1578](https://github.com/Microsoft/ChakraCore/pull/1578), so you might want to hold off on this?